src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -877,3 +877,48 @@ def _validator(namespace):
 
         setattr(namespace, property_name, service_types)
     return _validator
+
+
+def page_blob_tier_validator(namespace):
+    if not namespace.tier:
+        return
+
+    if namespace.blob_type != 'page' and namespace.tier:
+        raise ValueError('Blob tier is only applicable to page blobs on premium storage accounts.')
```

Comment:
It may change in the future and during test time the error message was not intuitive.